### ISSUE TITLE
Transform `Columns` into per-BoundedContext class

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#635]
Now the `Columns` class mixes utilities and cache of known entity columns.  It should be transformed into a class which is instantiated per `BoundedContext` instance.

Think of having a class that represents columns per entity, to avoid static utility methods here that pass data to each other.

The Javadoc for this class says:
```
Inherited columns are taken into account too, but building entity hierarchies is strongly discouraged.
```
We need to elaborate on why it is discouraged OR disable such a behaviour at all.

**Comment by alexander-yevsyukov** [comment]
@armiol, FYI.

**Comment by armiol** [comment]
Resolved in #681.
